module/Finna/src/Finna/RecordDriver/SolrFinnaTrait.php

```diff
@@ -65,12 +88,14 @@ public function getAccessRestrictions()
     /**
      * Return type of access restriction for the record.
      *
+     * @param string $language Language
+     *
      * @return mixed array with keys:
      *   'copyright'   Copyright (e.g. 'CC BY 4.0')
      *   'link'        Link to copyright info, see IndexRecord::getRightsLink
      *   or false if no access restriction type is defined.
      */
-    public function getAccessRestrictionsType()
+    public function getAccessRestrictionsType($language)
```

Comment:
Kielen voisi ottaa nykyisestä käyttökielestä, jos ei ole tarpeen jostain syystä näyttää jotain muuta, tähän tapaan:

`        list($locale) = explode('-', $this->getTranslatorLocale());`

Ks. SolrForward.php.

Reply:
Hmm, tätä kutsutaan getImageRights-metodista, joka saa jo nyt $language parametrin:
https://github.com/NatLibFi/NDL-VuFind2/blob/dev/module/Finna/src/Finna/RecordDriver/SolrFinnaTrait.php#L303

Meillä on ainakin pari tapaa käsitellä käyttäjän kielivalinta, sillä Loader asettaa driverille $preferredLanguage:n. Toiminta kannattaa yhtenäistää, mutta tehdäänkö eri PR:ssä? Toteutusta varten pitää myös miettiä voidaako kaikissa tapauksissa menetellä samalla tavalla kun käytäjän kielivalinnalla ei löydy näytettävää: riittääkö esim. tarkistaa järjestyksessä 
1. kielivalinta
2. näkymän oletuskieli
3. kieli joka on lueteltuna ensimmäisenä metadatassa ?

Reply:
Okei, no laita vaikka niin, että $language = null ja haetaan localesta, jos se on null.

Yhtenäistäminen on hyvä idea, mutta tehdään erikseen. Varmaan voisi olla preferredLanguages prioriteettijärjestyksessä tms. (nykyinen käyttökieli > oletuskieli > mitä nyt ekana löytyy).

Reply:
Ja uohdetaanpa koko juttu. Oli vähän sekava toteutus dev:ssä, nyt siivottu.